docs/tutorial/sha256_tutorial.ipynb

```diff
@@ -0,0 +1,768 @@
+        "print(\"Our SHA256:\", slices_to_hexarray(result))\n",
+        "print(\"Match:\", slices_to_hexarray(result)==m.hexdigest())"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 14,
+      "metadata": {
+        "id": "1uHN9GXgla_z"
+      },
+      "outputs": [],
+      "source": [
+        "class HomomorphicSHA:\n",
+        "    circuit: fhe.Circuit\n",
+        "    def __init__(self) -> None:\n",
+        "        inputset=[\n",
+        "            sha256_preprocess(np.random.randint(0, 2 ** 8, size=(150,)))\n",
+        "            for _ in range(100)\n",
+        "        ]\n",
+        "        # Compilation of the circuit should take a few minutes\n",
+        "        compiler = fhe.Compiler(sha256, {\"data\": \"encrypted\"})\n",
+        "        self.circuit = compiler.compile(\n",
+        "            inputset=inputset,\n",
+        "            configuration=fhe.Configuration(\n",
+        "                enable_unsafe_features=True,\n",
+        "                use_insecure_key_cache=True,\n",
+        "                insecure_key_cache_location=\".keys\",\n",
+        "                dataflow_parallelize=True,\n",
+        "            ),\n",
+        "            verbose=False,\n",
+        "        )\n",
+        "    \n",
+        "    def getSHA(self, data):\n",
+        "        return self.circuit.encrypt_run_decrypt(sha256_preprocess(data))\n",
+        "\n",
+        "    def getPlainSHA(self, data):\n",
+        "        return sha256(sha256_preprocess(data))\n",
+        "\n",
+        "sha = HomomorphicSHA()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "EkF0UxTcv_cQ"
+      },
+      "outputs": [],
+      "source": [
+        "# WARNING: This takes a LONG time\n",
+        "text = (\n",
+        "    b\"Lorem ipsum dolor sit amet, consectetur adipiscing elit. \"\n",
+        "    b\"Curabitur bibendum, urna eu bibendum egestas, neque augue eleifend odio, et sagittis viverra.\"\n",
+        ")\n",
+        "input_bytes = np.frombuffer(text, dtype=np.uint8)\n",
+        "encrypted_evaluation = sha.getSHA(input_bytes)\n",
+        "\n",
+        "print(\"Encrypted Evaluation: \", slices_to_hexarray(encrypted_evaluation))\n",
+        "print(\"    Plain Evaluation: \", slices_to_hexarray(sha.getPlainSHA(input_bytes)))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "O3FaAvxWr4EX"
+      },
+      "source": [
+        "### Smaller Example\n",
+        "The full sha256 function will take a long time to compile and execute. The function below only runs two iterations of the protocol. It can be used to demonstrate the function.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "g4eYdeRHrSy9"
+      },
+      "outputs": [],
+      "source": [
+        "def smaller_sha(data):\n",
```

Comment:
To avoid complicating the code, added a parameter for number of rounds. That can be changed to make the code faster. dc9457466136e0887b7d863d54f6fe93468b5395